### PR TITLE
refactor(core): Consolidate path-related errors in Sentry (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Keep reporting [path-related errors](https://n8nio.sentry.io/issues/4649493725) in Sentry but consolidate them in a single error group.

Also, add `options.extra` as `meta` so they remain visible in debug logs:

```
2023-11-24T11:50:54.852Z | error    | ReportableError: Something went wrong "{ test: 123, file: 'LoggerProxy.js', function: 'exports.error' }"
```

### CODE REVIEW

File: packages/core/src/errors.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 export class FileNotFoundError extends Error {
 	constructor(readonly filePath: string) {
-		super(`File not found: ${filePath}`);
+		super('File not found', { cause: { filePath } });
```
Comment (flipswitchingmonkey) [inline packages/core/src/errors.ts:3]: small issue: the debug log does not show the `cause` field currently, so the error will be missing some vital details there...

Comment (netroy) [inline packages/core/src/errors.ts:3]: I generally prefer not using non-errors as `cause` or in `throw`. Can we instead add a `filePath` property on the error, and then update the error-reporter to pick that property, and send it in `extras` ?

**Review by netroy — CHANGES_REQUESTED** [review]

**Review by netroy — DISMISSED** [review]

File: packages/cli/src/ErrorReporting.ts [hunk 2]
```diff
@@ -41,10 +41,19 @@ export const initErrorHandling = async () => {
 	addGlobalEventProcessor((event, { originalException }) => {
 		if (originalException instanceof ExecutionBaseError && originalException.severity === 'warning')
```
Comment (netroy) [inline packages/cli/src/ErrorReporting.ts:42]: we should (in another PR), rename `ReportableError` to `N8NError` or `ApplicationError`, and then update all these other errors to extend from that instead. Then we can also replace `severity` with `level` everywhere, and also start adding node/credential types in `tags`, and workflowId/executionId in extras.

**Review by netroy — APPROVED** [review]

File: packages/cli/src/ErrorReporting.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/ErrorReporting.ts:42]: Will do! Finally we have a basis to unify all this.

**Pull request merged**